app.go

```diff
@@ -287,20 +287,21 @@ func ErrorHook(funcs ...ErrorHandler) Option {
 	return errorHookOption(funcs)
 }
 
-// ErrorHandler implements Handle. They are used as error hooks and are called
-// on invoke errors.
+// ErrorHandler handles Fx application startup errors.
 type ErrorHandler interface {
 	HandleError(error)
 }
 
 type errorHookOption []ErrorHandler
 
+type errorHandlerList []ErrorHandler
+
```

Comment:
nit: move errorHandlerList next to its method, and errorHookOption next to its method.

We generally go for, type, constructor (if any), and methods when ordering them.